spyderlib/spyder.py

```diff
@@ -2653,6 +2661,59 @@ def start_open_files_server(self):
                 self.sig_open_external_file.emit(fname)
             req.sendall(b' ')
 
+    # ---- Quit and restart
+    def restart(self):
+        """Quit and Restart Spyder application"""
+        # Get start path to use in restart script
+        spyder_start_directory  = get_module_path('spyderlib')
+        spyder_folder = osp.split(spyder_start_directory)[0]
+        restart_script = osp.join(spyder_start_directory, 'spyder_restart.py')
+
+        # Get any initial argument passed when spyder was started
+        # Variables defined in bootstrap.py and spyderlib\start_app.py
+        env = os.environ.copy()
+        bootstrap_args = env.pop('SPYDER_BOOTSTRAP_ARGS', None)
+        spyder_args = env.pop('SPYDER_ARGS')
+
+        # Get current process and python running spyder
+        pid = os.getpid()
+        python = sys.executable
+
+        # Check if started with bootstrap.py
+        if bootstrap_args is not None:
+            spyder_args = bootstrap_args
+            is_bootstrap = True
+        else:
+            is_bootstrap = False
+
+        # Pass variables as environment variables (str) to restarter subprocess
+        env['SPYDER_ARGS'] = spyder_args
+        env['SPYDER_PID'] = str(pid)
+        env['SPYDER_IS_BOOTSTRAP'] = str(is_bootstrap)
+        env['SPYDER_FOLDER'] = spyder_folder
+
+        # Build the command and popen arguments depending on the OS
+        startupinfo = None
+        if os.name == 'nt':
+            startupinfo = subprocess.STARTUPINFO()
+            startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+            shell = False
+        else:
+            shell = True
+
+        command = '"{0}" "{1}"'
+        command = command.format(python, restart_script)
+
+        try:
+            if self.closing(True):

```

Comment:
I thought that `self.closing` closed the app, but I was mistaken. However, thanks for doing this change :-)